### PR TITLE
Update jspm package configuration

### PR DESCRIPTION
**Pull request by jdanyow**
Related to https://github.com/jspm/registry/pull/534

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -82,16 +82,18 @@
...
         "exports": "$"
       }
     },
-    "files": [
-      "css",
-      "fonts",
-      "js"
-    ]
+    "dependencies": {
+      "jquery": "github:components/jquery"
```
Comment (cvrebert) [inline package.json]: We can't use `jquery/jquery` or npm ?

Comment (jdanyow) [inline package.json]: This line tells jspm that bootstrap has a dep on jquery.  The fully qualified name for the jquery dep is `github:components/jquery`, which is consistent with what's in the [jspm registry](https://github.com/jspm/registry/blob/7171a47df3c0788b2add50710e68cd8f54f622c3/registry.json#L218).

CCing @guybedford in case I've got this wrong

Comment (cvrebert) [inline package.json]: I'd prefer not to rely on some third-party packaging like `components` if we don't have to.

Comment (jdanyow) [inline package.json]: ok- that makes sense.  Let me see if I can find out why it's configured that way in the jspm registry.

Comment (guybedford) [inline package.json]: To use jspm's default jquery, whatever that is, try setting `"jquery": "*"` (or the jQuery version range - `"jquery": "^2.1.4"`). This way if we change the main jquery source package to something more official in due course, Bootstrap would naturally track that.

Comment (cvrebert) [inline package.json]: Ok, so then it should be `"jquery": "1.9.1 - 2"` just like in our `bower.json`.

Comment (guybedford) [inline package.json]: Sure, unfortunately we don't currently support arbitrary version ranges (just semver and tilde), so it may be best to stick with `*` for jQuery in this case.

Comment (cvrebert) [inline package.json]: > we don't currently support arbitrary version ranges

Out of curiosity, what's the reason for that? You don't use [node-semver](https://github.com/npm/node-semver)?

**Comment by jdanyow** [comment]
updated jquery dependency per discussion :point_up:

File: package.json (hunk 1 above)
Comment (guybedford) [inline package.json]: Sure, there might be benefits to being able to resolve version ranges in the browser, which is made much easier by sticking to simple ranges (can replace node-semver with the much-smaller https://github.com/systemjs/systemjs/blob/0.16.11/lib/extension-versions.js). Still seeing how it works out, but so far haven't found it to be too restricting that this limitation has been critical yet.

File: package.json [hunk 2]
```diff
@@ -82,16 +82,18 @@
   ],
   "jspm": {
     "main": "js/bootstrap",
+    "ignore": ["dist/js/npm"],
+    "directories": {
+      "lib": "dist"
```
Comment (cvrebert) [inline package.json]: Since we now use ES6, would it be better to set `"format": "es6"` and *not* use the `dist` JavaScript?

Comment (jdanyow) [inline package.json]: I can't think of a solid reason people would want the raw ES6 files- did you have a use case in mind?

People will want the ability to load the individual bootstrap components- which dist/js/umd/... enables.

Comment (cvrebert) [inline package.json]: [jspm](http://jspm.io/) supports ES6 modules:
> * jspm is a package manager for the SystemJS universal module loader, built on top of the dynamic ES6 module loader
> 
> * Load any module format (ES6, ...

Comment (jdanyow) [inline package.json]: Not sure what your point is- I'll defer to you and @guybedford on this of course- let me know if you guys want me to amend the PR.

Let me clarify what I meant here:

>I can't think of a solid reason people would want the raw ES6 files- did you have a use case in mind?

jspm supports a variety of module formats, which makes it very easy to consume libraries, without concern for what format they're distributed in.  I think the only time I would be concerned is if the library was distributed in ES6/ES7 format and I had to transpile it as part of my build process (or worse, using the loader).

In other words I think it's an edge case for someone to say *"oh man I'm a super cutting edge guy and I only use ES6/ES7 modules (except for that pesky jQuery dependency which I'm ok with)".*  

Does this make sense?  If there is a compelling reason to distribute the ES6 bits, you might consider taking the approach the Aurelia repos take where the dist folder has subfolders for the amd, commonjs, es6 and system formats.

Thoughts?

Comment (guybedford) [inline package.json]: Yes it's best to use the existing dist folder - for example lots of users use jspm without any transpiler enabled and just to manage ES5 module formats. It's still the lowest common denominator for a little while yet.

**Comment by cvrebert** [comment]
Thanks! Sorry for the delay.
